apps/src/lib/ui/checkedRadioButton.css

```diff
@@ -0,0 +1,34 @@
+/*
+Basis for this CSS from:
+https://weblog.west-wind.com/posts/2015/Feb/26/Using-FontAwesome-Fonts-for-HTML-Radio-Buttons-and-Checkboxes
+*/
+
+input[type=radio].with-font {
```

Comment:
Nice work finding and implementing this approach! I didn't know we could import CSS like this into a JS file.

I'm a bit worried that this deviates from our existing patterns for how we store our CSS, though. Ideally, if Radium can handle this syntax, this would live as inline CSS within the react component. If that's not possible though then the next best place for this is in https://github.com/code-dot-org/code-dot-org/blob/staging/apps/style/common.scss

CC @islemaster for any other opinions how how we serve CSS.

Reply:
I actually suggested this approach - we already use it in a few places for third-party React components that expect certain CSS to be present on the page, like `react-virtualized-select`.

https://github.com/code-dot-org/code-dot-org/blob/1cad5ac667f9ed1a3aa31063e6d06bd21a78e2bd/apps/src/templates/SchoolAutocompleteDropdown.jsx#L4-L6

Radium [doesn't support the `:before` and `:after` pseudo-selectors.](https://github.com/FormidableLabs/radium/tree/master/docs/faq#how-do-i-use-pseudo-selectors-like-checked-last-before-or-after) I'm fine with this approach to very component-specific CSS because I think it solves the same problem that inlining our CSS in React is solving - bundling relevant styles with the components that use them, so they look the same no matter what CSS-context they are loaded into.

Totally true this isn't something we've discussed thoroughly as a team though - interested in talking through pros and cons.  I wouldn't be surprised if there's some duplication of CSS with this approach (but probably no more than when inlining it into React) - also not sure if there are performance implications, or pain from having two possible ways to transpile SCSS, etc.

Reply:
Thanks for the background info @islemaster. This solution seems "safe" in the sense that it does not seem to have any of the potential problems I can think of -- the changes should be confined to this page, seem like they'd keep working even after we minify javascript, and wouldn't have any fingerprinting problems because the js file you're embedding the css into is fingerprinted.

Is there a particular reason this CSS needs to be in the same page as the JS, or would inclusion in common.scss work? I think I'm just resistant to having a separate way to do things without clearly understanding when to use one vs the other.

If the CSS really does need to be in the same page, I'm fine with saying to use this method when you need that and to use *.scss when a separate stylesheet file would work instead.

Reply:
I don't quite follow "in the same page;" do you mean is it a requirement that this CSS _only_ be included when the relevant component is present?  It's global-ish either way and I think there's some risk involved with either approach, but that all depends on constructing our CSS well.  I like this approach because there's a bit less action-at-a-distance going on and keeping single-use CSS close to its consumer seems good.  I agree that reusable CSS makes more sense included in a top-level stylesheet.

Reply:
OK, i see what you're saying -- the thing we're trying to implement here would totally work as part of common.scss, however this approach makes it limited to the pages which need it whereas using common.scss is more like adding it globally.

I think we both understand each other now, but to answer your question anyway, "In the same page" was just my attempt to paraphrase this comment of yours:

> we already use it in a few places for third-party React components that expect certain CSS to be present on the page